orchestration/dagster_orchestration/hca_utils/utils.py

```diff
@@ -255,25 +259,28 @@ def process_duplicates(self, soft_delete: bool = False):
         Check and print the number of duplicates for each table in the dataset.
         :return:
         """
-        self._process_rows(self.get_all_table_names, self.get_duplicates, soft_delete=soft_delete, issue="duplicate rows")
+        return self._process_rows(self.get_all_table_names, self.get_duplicates, soft_delete=soft_delete,
+                                  issue="duplicate rows")
 
     def process_null_file_refs(self, soft_delete: bool = False):
         """
         Check/remove and print the number of null file references for each table in the dataset that has a `file_id`
         column.
         :return:
         """
-        self._process_rows(self.get_file_table_names, self.get_null_filerefs, soft_delete=soft_delete, issue="null file refs")
+        return self._process_rows(self.get_file_table_names, self.get_null_filerefs, soft_delete=soft_delete,
+                                  issue="null file refs")
 
     def check_for_all(self):
         """
         Check and print the number of duplicates and null file references in all tables in the dataset.
         :return:
         """
         logging.info("Processing...")
-        self.process_duplicates()
-        self.process_null_file_refs()
+        duplicate_count = self.process_duplicates()
+        null_file_ref_count = self.process_null_file_refs()
         logging.info("Finished.")
+        return duplicate_count + null_file_ref_count
```

Comment:
❓ What if we return a richer object, like a NamedTuple that contains the dupes + nulls broken out into fields? I generally favor finer-grained values like that for diagnostics and aggregating only when necessary.

Reply:
I like it, will change